Source/WebCore/platform/network/cf/AuthenticationChallenge.h

```diff
@@ -36,10 +36,16 @@ namespace WebCore {
 class AuthenticationChallenge : public AuthenticationChallengeBase {
 public:
     AuthenticationChallenge() {}
+    WEBCORE_EXPORT AuthenticationChallenge(const AuthenticationChallenge&);
```

Comment:
These changes are surprising but seem fine.  Why does this structure require a change but others don't?

Reply:
From the commit log message:

```
* Source/WebCore/platform/network/cf/AuthenticationChallenge.h:
* Source/WebCore/platform/network/mac/AuthenticationMac.mm:
(WebCore::AuthenticationChallenge::AuthenticationChallenge): Add.
(WebCore::AuthenticationChallenge::operator=): Add.
- Implement the "Rule of Five" for AuthenticationChallenge.  The
  change to leakRef() caused build failures when implicit move
  constructors and copy assignment operators were used in C++
  code because the instance variables were RetainPtr<> objects
  holding Cocoa types.
```

Reply:
Interestingly, the [ios](https://ews-build.webkit.org/#/builders/26/builds/5573) and [ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5802) build failures appear to be due to lack of support for `RetainPtr<id>` in pure C++ code.

I guess that would use `CF_RETURNS_RETAINED` because it basically can't use `NS_RETURNS_RETAINED`.

Reply:
And it seems that since I implemented a _proper_ move constructor/move assignment operator that empty the values in the moved-from object, I may have triggered some latent use-after-move bugs based on the [mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4962l) crashes.